hfgl/model.py

```diff
@@ -469,11 +471,43 @@ def __init__(self, config: dict | VocoderConfig):
         self.config = config
         self.generator = Generator(self.config)
 
+    def check_and_upgrade_checkpoint(self, checkpoint):
+        """
+        Check model's compatibility and possibly upgrade.
+        """
+        model_info = checkpoint.get(
+            "model_info",
+            {
+                "name": self.__class__.__name__,
+                "version": "1.0",
+            },
+        )
+
+        ckpt_model_type = model_info.get("name", "MISSING_TYPE")
+        if ckpt_model_type != self.__class__.__name__:
+            raise TypeError(
+                f"""Wrong model type ({ckpt_model_type}), we are expecting a '{ self.__class__.__name__ }' model"""
+            )
+
+        ckpt_version = model_info.get("version", "0.0")
+        if ckpt_version > self._VERSION:
+            raise ValueError(
+                "Your model was created with a newer version of EveryVoice, please update your software."
+            )
+        # Successively convert model checkpoints to newer version.
+        if ckpt_version < "1.0":
+            # TODO: Write code to convert model to version 1.0.
```

Comment:
Very picky comment here: this is not "TODO", because we know converting from version 0 to 1.0 requires no changes. I might say `# Upgrading from 0.0 to 1.0 requires not changes` instead so we don't leave a misleading "TODO" flag in the code.
The TODO will be for when we write version 2.0, to add conversion code for that, and I think the comment on line 497 is already sufficiently clear for that.